terminal.cpp

```diff
@@ -213,18 +219,111 @@ void Terminal::keyPress(int key, int modifiers)
         return;
     }
 
-    char cursorModif='[';
-    if(iAppCursorKeys)
-        cursorModif = 'O';
+    int modcode = (modifiers & Qt::ShiftModifier ? 1 : 0) | 
+                  (modifiers & Qt::AltModifier ? 2 : 0) |
+                  (modifiers & Qt::ControlModifier ? 4 : 0);
+
+    if (modcode == 0) {
+        char cursorModif='[';
+        if(iAppCursorKeys)
+            cursorModif = 'O';
+
+        if( key==Qt::Key_Up )
+            toWrite += QString("%1%2A").arg(ch_ESC).arg(cursorModif).toLatin1();
+        if( key==Qt::Key_Down )
+            toWrite += QString("%1%2B").arg(ch_ESC).arg(cursorModif).toLatin1();
+        if( key==Qt::Key_Right )
+            toWrite += QString("%1%2C").arg(ch_ESC).arg(cursorModif).toLatin1();
+        if( key==Qt::Key_Left )
+            toWrite += QString("%1%2D").arg(ch_ESC).arg(cursorModif).toLatin1();
+        if( key==Qt::Key_PageUp )
+            toWrite += QString("%1[5~").arg(ch_ESC).toLatin1();
```

Comment:
On earlier code above CSI was formed with ch_ESC and cursorModif. I have no idea why the name is cursorModif in the first place, but could expect for consistency control codes being formed the same way here. Better might be combining those two characters separately instead of individually for each case.